aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -8,7 +8,36 @@
 
 void runMPSGraph(MPSStream* mpsStream, MPSGraph* mpsGraph, NSDictionary* feeds,
                  NSDictionary* results, bool disableTypeInference) {
-  mpsStream->executeMPSGraph(mpsGraph, feeds, results, SyncType::COMMIT_ADAPTIVE, disableTypeInference);
+  mpsStream->executeMPSGraph(mpsGraph, feeds, results, SyncType::COMMIT_ADAPTIVE);
+}
+
+// this should be merged into runMPSGraph() with new arg "disableTypeInference" for executables
+void runMPSGraphExecutable(MPSStream *mpsStream, MPSCachedGraph *cachedGraph, NSDictionary *feeds, NSDictionary *results, MPSGraphTensor* alphaTensor) {
+  @autoreleasepool {
+    MPSGraph *mpsGraph = cachedGraph->graph();
+    MPSGraphExecutable* executable = cachedGraph->getExecultable();
+    if (!executable) {
+      NSMutableDictionary* shapes = [[NSMutableDictionary new] autorelease];
+      for (MPSGraphTensor* graphTensor in feeds) {
+        MPSGraphTensorData* graphTensorData = [feeds objectForKey:graphTensor];
+        shapes[graphTensor] = [[[MPSGraphShapedType alloc] initWithShape:nil dataType:graphTensorData.dataType] autorelease];
+      }
+      if (alphaTensor) {
+        shapes[alphaTensor] = [[[MPSGraphShapedType alloc] initWithShape:@[@1] dataType:alphaTensor.dataType] autorelease];
```

Comment:
This is needed for the alpha tensor ranked placeholder which is always 1. I will update that tensor to be unranked when type inference is disabled.